apps/s_client.c

```diff
@@ -3034,6 +3034,23 @@ int s_client_main(int argc, char **argv)
      * TODO: Find a better way to do this
      */
     Sleep(50);
+#endif
```

Comment:
Reiterating this part...  it needs to be of this form, I think:

``` C
#if defined(OPENSSL_SYS_WINDOWS)
    Sleep(50);
#elif defined(OPENSSL_SYS_CYGWIN)
    usleep(50000);
#endif
```

Note: I'm not done testing, so hold this thought for a bitt

Reply:
I saw your note on making a separate PR for this.  Ok, I will.

Reply:
do we need the usleep and the shutdown for cygwin?

Reply:
Yes.  Cygwin is a POSIX library, so it provides a POSIX API rather than a Windows API.

Regarding `shutdown`, you will therefore also have the API that you recognise from Unix.  And the compiler does *not* defined `_WIN32`, so no worries there